pkg/manager/member/tidb_member_manager.go

```diff
@@ -448,6 +448,7 @@ func getNewTiDBServiceOrNil(tc *v1alpha1.TidbCluster) *corev1.Service {
 	tidbSelector := label.New().Instance(instanceName).TiDB()
 	svcName := controller.TiDBMemberName(tcName)
 	tidbLabels := util.CombineStringMap(tidbSelector.Copy().UsedByEndUser().Labels(), svcSpec.Labels)
+	tidbSelectorLabels := util.CombineStringMap(tidbSelector.Copy().Labels(), svcSpec.SelectorLabels)
```

Comment:
Why do you need to update the selector? Or you actually need to update the Pod labels?